### PR TITLE
Added: Spider for detecting broken links of a given website

### PR DESCRIPTION
**Pull request by starrify**

### CODE REVIEW

File: testspiders/spiders/broken_link.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            netloc = urlparse.urlparse(input_url).netloc
+            domain = netloc.split('@')[-1].split(':')[0]
+            self.allowed_domains = [domain]
+
+    def parse(self, response):
+        """Parses a default response"""
+        if not isinstance(response, scrapy.http.TextResponse):
+            self.crawler.stats.inc_value('binary_response')
+            return
+        if response.status >= 400 and response.status <= 500:
```
Comment (akarthik86) [inline testspiders/spiders/broken_link.py]: why not 5xx not considered as broken links ?

File: testspiders/spiders/broken_link.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# coding: utf8
```
Comment (akarthik86) [inline testspiders/spiders/broken_link.py]: I would prefer inline docuemntation explaining the spider. ( can bring the readme content here ) and also inline documentation on what spider is trying to do at different decision points.

Comment (starrify) [inline testspiders/spiders/broken_link.py]: The change has been made.

**Comment by redapple** [comment]
@starrify ,
I believe the spider should also handle DNS errors (or unresponsive servers even)

**Comment by starrify** [comment]
Hi @redapple thanks for the hint! Yes it should have been able to do that. I'll update this PR to make it possible. :)

**Comment by redapple** [comment]
@starrify , any updates ?

**Comment by starrify** [comment]
Hi @redapple , the PR has been updated with a few changes.

**Pull request merged**